mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/DatasourceAttributeExtractor.scala

```diff
@@ -34,16 +40,117 @@ private[autologging] object DatasourceAttributeExtractor {
     }
   }
 
+  protected def maybeGetDeltaTableInfo(plan: LogicalPlan): Option[SparkTableInfo]
+
   /**
    * Get SparkTableInfo representing the datasource that was read from leaf node of a Spark SQL
    * query plan
    */
-  def getTableInfoToLog(leafNode: LogicalPlan): Option[SparkTableInfo] = {
+  protected def getTableInfoToLog(leafNode: LogicalPlan): Option[SparkTableInfo] = {
+    val deltaInfoOpt = maybeGetDeltaTableInfo(leafNode)
+    if (deltaInfoOpt.isDefined) {
+      deltaInfoOpt
+    } else {
+      leafNode match {
+        case relation: DataSourceV2Relation =>
+          getSparkTableInfoFromTable(relation.table)
+        case other =>
+          None
+      }
+    }
+  }
+
+  private def getLeafNodes(lp: LogicalPlan): Seq[LogicalPlan] = {
+    if (lp == null) {
+      return Seq.empty
+    }
+    if (lp.isInstanceOf[LeafNode]) {
+      Seq(lp)
+    } else {
+      lp.children.flatMap { child =>
+        child match {
+          case l: LeafNode =>
+            Seq(l)
+          case other: LogicalPlan => getLeafNodes(other)
+        }
+      }
+    }
+  }
+
+  /**
+   * Get SparkTableInfo representing the datasource(s) that were read from a SparkListenerEvent
+   * assumed to have a QueryExecution field named "qe".
+   */
+  def getTableInfos(event: SparkListenerSQLExecutionEnd): Seq[SparkTableInfo] = {
+    val qe = SparkAutologgingUtils.getQueryExecution(event)
+    if (qe != null) {
+      val leafNodes = getLeafNodes(qe.analyzed)
+      leafNodes.flatMap(getTableInfoToLog)
+    } else {
+      Seq.empty
+    }
+  }
+}
+
+object DatasourceAttributeExtractor extends DatasourceAttributeExtractorBase {
```

Comment:
Default datasource attribute extractor (TODO sid to add a comment clarifying this)